examples/MomentoApplication/Program.cs

```diff
@@ -17,25 +17,50 @@ class Program
         async static Task Main(string[] args)
         {
             using SimpleCacheClient client = new SimpleCacheClient(MOMENTO_AUTH_TOKEN, DEFAULT_TTL_SECONDS);
-            try
-            {
-                client.CreateCache(CACHE_NAME);
+
+            Console.WriteLine("Creating cache with an illegal empty name");
+            CreateCacheResponse emptyCreateResp = client.CreateCache("");
+            if (emptyCreateResp is CreateCacheResponse.Error error) {
+                Console.WriteLine($"Exception error code: {error.Exception.ErrorCode}");
+                Console.WriteLine($"Exception msg: {error.Exception.Message}");
             }
-            catch (AlreadyExistsException)
-            {
-                Console.WriteLine($"Cache with name {CACHE_NAME} already exists.\n");
+
+            Console.WriteLine($"Creating cache {CACHE_NAME}");
+            CreateCacheResponse createResp = client.CreateCache(CACHE_NAME);
+            if (createResp is CreateCacheResponse.Error errorResp2) {
+                if (errorResp2.Exception.ErrorCode == MomentoErrorCode.ALREADY_EXISTS_ERROR) {
```

Comment:
looking at this - do you think it would be possible to move the `ErrorCode` up one level?  Just thinking about how I'd *like* to write this, as a user:

```
if (createResp is CreateCacheResponse.Error) {
   if (createResp.ErrorCode == MomentoErrorCode.ALREADY_EXISTS_ERROR) {
      ...
   }
}
```

Also I'm guessing from your code here that if you want the C# compiler to do the smart cast then you have to give it a new variable name as you did with `errorResp2` above?  How did that feel to you when coding?  I am curious if that's a common idiom for C# devs (I honestly have no idea :) )

Reply:
The error code is added into the mix on a per-exception-type basis when the `SdkException` subclasses are constructed, so I left that logic as-is and added a getter to the response object. The getter just grabs and returns the code from the exception.

Fixed in a6184ba5

I'm still playing with the example code, but I did seem to have to use new names every time I did one of those casts. It was a little annoying, but there may be workarounds and I think the extra work is worth it for the ease of matching and casting.